packages/components/src/icon/index.js

```diff
@@ -8,24 +8,24 @@ import { cloneElement, createElement, Component, isValidElement } from '@wordpre
  */
 import { Dashicon, SVG } from '../';
 
-function Icon( { icon = null, size, className } ) {
+function Icon( { icon = null, size, className, ...additionalProps } ) {
 	let iconSize;
 
 	if ( 'string' === typeof icon ) {
 		// Dashicons should be 20x20 by default
 		iconSize = size || 20;
-		return <Dashicon icon={ icon } size={ iconSize } className={ className } />;
+		return <Dashicon icon={ icon } size={ iconSize } className={ className } { ...additionalProps } />;
 	}
 
 	// Any other icons should be 24x24 by default
 	iconSize = size || 24;
 
 	if ( 'function' === typeof icon ) {
 		if ( icon.prototype instanceof Component ) {
-			return createElement( icon, { className, size: iconSize } );
+			return createElement( icon, { className, additionalProps, size: iconSize } );
```

Comment:
We should be spreading `additionalProps` here, it's not a props name

Reply:
Done